docs/fundamentals/app-host-overview.md

```diff
@@ -97,28 +97,50 @@ Project-to-project references are handled differently than resources that have w
 
 | Method | Environment variable |
 |--|--|
-| `WithReference(cache)` | `ConnectionStrings__cache="localhost:6379"` |
-| `WithReference(apiservice)` | `services__apiservice__0="http://_http.localhost:8034"` <br /> `services__apiservice__1="http://localhost:8034"` |
+| `WithReference(cache)` | `ConnectionStrings__cache="localhost:62354"` |
+| `WithReference(apiservice)` | `services__apiservice__http__0="http://localhost:5455"` <br /> `services__apiservice__https__0="https://localhost:7356"` |
 
 Adding a reference to the "apiservice" project results in service discovery environment variables being added to the front-end. This is because typically, project to project communication occurs over HTTP/gRPC. For more information, see [.NET Aspire service discovery](../service-discovery/overview.md).
 
 It's possible to get specific endpoints from a container or executable using the <xref:Aspire.Hosting.ResourceBuilderExtensions.WithEndpoint%2A> and calling the <xref:Aspire.Hosting.ResourceBuilderExtensions.GetEndpoint%2A>:
 
 ```csharp
 var customContainer = builder.AddContainer("myapp", "mycustomcontainer")
-                             .WithEndpoint(containerPort: 9043, name: "endpoint");
+                             .WithHttpEndpoint(port: 9043, name: "endpoint");
 
 var endpoint = customContainer.GetEndpoint("endpoint");
 
 var apiservice = builder.AddProject<Projects.AspireApp_ApiService>("apiservice")
                         .WithReference(endpoint);
 ```
 
-| Method                    | Environment variable                            |
-|---------------------------|-------------------------------------------------|
-| `WithReference(endpoint)` | `services__myapp__0=endpoint://localhost:65256` |
+| Method                    | Environment variable                                  |
+|---------------------------|-------------------------------------------------------|
+| `WithReference(endpoint)` | `services__myapp__endpoint__0=https://localhost:9043` |
 
-The `containerPort` parameter is the port that the container is listening on. For more information on container ports, see [Container ports](networking-overview.md#container-ports). For more information on service discovery, see [.NET Aspire service discovery](../service-discovery/overview.md).
+The `port` parameter is the port that the container is listening on. For more information on container ports, see [Container ports](networking-overview.md#container-ports). For more information on service discovery, see [.NET Aspire service discovery](../service-discovery/overview.md).
+
+### Service endpoint environment variable format
+
+In the preceding section, the `WithReference` method is used to express dependencies between resources. When service endpoints result in environment variables being injected into the dependent resource, the format may not be obvious. The following section provides details on this format.
```

Comment:
>The following section provides details on this format.

Should this say "this section" rather than "the following section"? The following section is **APIs for adding and expressing resources**.